server/routes/prisons/sessionTemplates/addSessionTemplateController.ts

```diff
@@ -88,6 +88,9 @@ export default class AddSessionTemplateController {
         locationGroupReferences = locationGroups.map(locationGroup => locationGroup.reference)
       }
 
+      const publicClient = sessionTemplate.clients.find(client => client.userType === 'PUBLIC')
```

Comment:
Is it possible that `publicClient` here could be `undefined` - i.e. there isn't an entry in `clients` with `userType` of `PUBLIC`? Perhaps for old session template data prior to back-end changes for supporting this? If so, next line will fail.

Could mitigate this with `publicClient?.active === true ? ...`

Reply:
from the testing i've done, all session templates have a public client - assuming the back end pre-populated these when they created this new field, but I'll see if I can make a catch

Reply:
Never assume 😄

Reply:
Can we move this into a function, lines 91/92 ? It's used in 3 places.